### ISSUE TITLE
Proposal: add verbose logging feature

### ISSUE THREAD

**Issue by mc1098** [mc1098/sap#71]
It would be nice, partly for my own sanity, to have a feature to turn on that is very verbose about what sap is doing at each point. Comparing the two elements accessibility names as it iterates over the elements with the correct Role and even when a generic match fails. Some of these things might be obvious when you are used to this environment but won't be for a lot.

Pros: makes it easier to inspect what sap does
Cons: will make the code much more verbose and harder to change
